src/map/script.c

```diff
@@ -7933,6 +7933,90 @@ BUILDIN(makeitem)
 	return true;
 }
 
+/*==========================================
+* makeitem2 <item id>,<amount>,<identify>,<refine>,<attribute>,<card1>,<card2>,<card3>,<card4>,{"<map name>",<X>,<Y>,<range>};
+*------------------------------------------*/
+BUILDIN(makeitem2)
+{
+	struct map_session_data *sd = NULL;
+	struct item_data *i_data;
+	int nameid = 0, amount;
+	int16 x, y, m = -1, range;
+	struct item item_tmp;
+
+	if (script_isstringtype(st, 2)) {
+		const char *name = script_getstr(st, 2);
+		struct item_data *item_data = itemdb->search_name(name);
+		if (item_data == NULL)
+			nameid = item_data->nameid;
+	} else {
+		nameid = script_getnum(st, 2);
+	}
+
+	i_data = itemdb->exists(nameid);
+	if (i_data == NULL) {
+		ShowError("makeitem2: Unknown item %d requested.\n", nameid);
+		return true;
+	}
+
+	if (script_hasdata(st, 11)) {
+		m = map->mapname2mapid(script_getstr(st, 11));
+	} else {
+		sd = script->rid2sd(st);
```

Comment:
if moved to top, it would caused to script to run only when a player is attached.
why need to force attach the player when it's optional ?